### ISSUE TITLE
Assets not served

### ISSUE THREAD

**Issue by davidterranova** [neckhair/rails-on-docker#2]
Hi,
I am trying your system and the assets are not served by Nginx.
I have the following error in the logs:

`web_1    | X-Accel-Mapping header missing`

It seems that the Nginx proxy does not handle the `X-Accel-Redirect` header.

Best regards,

**Comment by neckhair** [comment]
I can't reproduce the problem. Did you precompile the assets?

**Comment by neckhair** [comment]
Just ran it again on a clean environment. All works fine.

**Comment by reejosamuel** [comment]
I really like your setup but I think the assets not being served or the least not correctly.

Nginx needs to be told where the static assets are using this config `root /rails/public;` but since you are using `jwilder/nginx-proxy` directly, this configuration is missing.

Moreover the nginx can't access the rails assets without linking volumes i feel. Please share you thoughts.

**Comment by brunocascio** [comment]
How to solve this?

**Comment by neckhair** [comment]
@reejosamuel You're right. I'll add it as a todo to the list. Maybe you already thought about a solution?

**Comment by brunocascio** [comment]
@neckhair I'll write an example later, using `Rails + MariaDB + jwilder/nginx-proxy.`

**Comment by neckhair** [comment]
It's possible to change the virtual host's config in `jwilder/nginx-proxy`. https://github.com/jwilder/nginx-proxy#per-virtual_host. So we can just use that to link to the rails container and get the assets there.

**Comment by brunocascio** [comment]
Yes but you don't have to serve the static files with rails. You should use nginx-proxy for that, so you need mount a data volume inside the nginx-proxy container for serve them.

**Comment by neckhair** [comment]
Yep, with "link" I meant using volumes.
I'll try it later but I'm not sure when I'll have the time for it.

**Comment by sabman** [comment]
@neckhair looks like you have your assets being served by Rails

https://github.com/neckhair/rails-on-docker/blame/master/config/environments/production.rb#L25

Kinda defeats the purpose of having nginx, no?

**Comment by neckhair** [comment]
@sabman Nginx serves as proxy and load balancer to the (many) Rails containers. So it has its purpose. But I agree that Nginx should serve the assets directly. (See discussion above.)

**Comment by reejosamuel** [comment]
@neckhair From what i remember, I worked around this issue by making the assets folder a shared volume on the rails container and referencing/linking it from the nginx container in my docker-compose file. this way nginx was serving the up to date assets from rails container without copy pasting and other hacky workarounds. Hope this helps
